src/DurableTask.Core/TaskOrchestrationContext.cs

```diff
@@ -484,6 +484,12 @@ public void HandleTimerFiredEvent(TimerFiredEvent timerFiredEvent)
             {
                 LogDuplicateEvent("TimerFired", timerFiredEvent, taskId);
             }
+
+            if (!this.IsReplaying)
+            {
+                Activity activity = TraceHelper.CreateActivityForTimer(this.OrchestrationInstance, timerFiredEvent.Timestamp, timerFiredEvent.FireAt);
```

Comment:
If you add `using` here you can remove the `activity.Stop()`. The disposal will call `Stop()` for you. Plus, `activity` might be `null` - the using will handle that case for you.

```suggestion
                using Activity activity = TraceHelper.CreateActivityForTimer(this.OrchestrationInstance, timerFiredEvent.Timestamp, timerFiredEvent.FireAt);
```